advanced/Scripts/list.sh

```diff
@@ -88,175 +88,144 @@ HandleOther() {
-            RemoveDomain "${dom}" "${listAlt}"
+            AddDomain "${dom}" "${listType}"
+            if [[ ! "${listType}" == "regex" ]]; then
+                RemoveDomain "${dom}" "${listAlt}"
+            fi
         else
-            RemoveDomain "${dom}" "${listMain}"
+            RemoveDomain "${dom}" "${listType}"
         fi
   done
 }
 
 AddDomain() {
+    local domain list num
+    domain="$1"
     list="$2"
-    domain=$(EscapeRegexp "$1")
-
-    [[ "${list}" == "${whitelist}" ]] && listname="whitelist"
-    [[ "${list}" == "${blacklist}" ]] && listname="blacklist"
-
-    if [[ "${list}" == "${whitelist}" || "${list}" == "${blacklist}" ]]; then
-        [[ "${list}" == "${whitelist}" && -z "${type}" ]] && type="--whitelist-only"
-        [[ "${list}" == "${blacklist}" && -z "${type}" ]] && type="--blacklist-only"
-        bool=true
-        # Is the domain in the list we want to add it to?
-        grep -Ex -q "${domain}" "${list}" > /dev/null 2>&1 || bool=false
-
-        if [[ "${bool}" == false ]]; then
-            # Domain not found in the whitelist file, add it!
-            if [[ "${verbose}" == true ]]; then
-                echo -e "  ${INFO} Adding ${1} to ${listname}..."
-            fi
-            reload=true
-            # Add it to the list we want to add it to
-            echo "$1" >> "${list}"
-        else
-            if [[ "${verbose}" == true ]]; then
-                echo -e "  ${INFO} ${1} already exists in ${listname}, no need to add!"
-            fi
-        fi
-    elif [[ "${list}" == "${regexlist}" ]]; then
-        [[ -z "${type}" ]] && type="--wildcard-only"
-        bool=true
-        domain="${1}"
 
-        [[ "${wildcard}" == true ]] && domain="(^|\\.)${domain//\./\\.}$"
+    # Is the domain in the list we want to add it to?
+    num="$(sqlite3 "${gravityDBfile}" "SELECT COUNT(*) FROM ${list} WHERE ${sqlitekey} = \"${domain}\";")"
 
-        # Is the domain in the list?
-        # Search only for exactly matching lines
-        grep -Fx "${domain}" "${regexlist}" > /dev/null 2>&1 || bool=false
-
-        if [[ "${bool}" == false ]]; then
-            if [[ "${verbose}" == true ]]; then
-                echo -e "  ${INFO} Adding ${domain} to regex list..."
-            fi
-            reload="restart"
-            echo "$domain" >> "${regexlist}"
-        else
-            if [[ "${verbose}" == true ]]; then
-                echo -e "  ${INFO} ${domain} already exists in regex list, no need to add!"
-            fi
+    if [[ "${num}" -eq 0 ]]; then
+        # Domain not found in the file, add it!
+        if [[ "${verbose}" == true ]]; then
+            echo -e "  ${INFO} Adding ${1} to ${listname}..."
+        fi
+        reload=true
+        # Add it to the list we want to add it to
+        local timestamp
+        timestamp="$(date --utc +'%s')"
+        sqlite3 "${gravityDBfile}" "INSERT INTO ${list} (${sqlitekey},enabled,date_added) VALUES (\"${domain}\",1,${timestamp});"
```

Comment:
The defaults for `enabled` and `date_added` in all of the tables are true and the current timestamp, respectively, so these columns can be omitted from the insert statement. This will also simplify the code.

Reply:
Leaving off `enabled` is fine, however, when simplifying this string to
```
domain="abc.de"
"INSERT INTO ${list} (${sqlitekey}) VALUES (\"${domain}\");"
```
it results in an inserted line
```
abc.de|1|2019-04-26 14:28:08|
```
where the `date_added` field is filled with a ISO8601 (UTC) datetime *string* instead of a numeric timestamp. This is allowed by the SQLite specifications (the type of `DATETIME` columns can be `INTEGER`, `TEXT`, or even `REAL`), however, as it consumes extra space, I will keep the injection of a numerical timestamp obtained calling `date` if you have no objections and only omit the `enabled = 1` here.

Note: Even when defining the column as `date_added INTEGER NOT NULL DEFAULT CURRENT_TIMESTAMP` in the `CREATE TABLE` (not the `INTEGER` type!), the injection command will still store a `TEXT` object in the `date_added` column...
```
sqlite> CREATE TABLE www (domain TEXT UNIQUE NOT NULL, enabled BOOLEAN NOT NULL DEFAULT 1, date_added INTEGER NOT NULL DEFAULT CURRENT_TIMESTAMP, comment TEXT);
sqlite> insert into www (domain) values ("abc");
sqlite> select * from www;
abc|1|2019-04-26 14:37:21|
```

Reply:
It looks like we can have the best of both worlds, at least for the default:
https://stackoverflow.com/a/11556717

Storing as integer is fine by me, it would narrow the allowed types to what we want

Reply:
@dl6er it looks like we also need to cast to integer, as that function will return the number as a string: https://stackoverflow.com/a/29420016